includes/data-stores/class-wc-admin-reports-coupons-data-store.php

```diff
@@ -350,14 +350,12 @@ protected function get_cache_key( $params ) {
 	 * Create or update an an entry in the wc_order_coupon_lookup table for an order.
 	 *
 	 * @since 3.5.0
-	 * @param int $order_id Order ID.
+	 * @param object $order Order.
 	 * @return int|bool Returns -1 if order won't be processed, or a boolean indicating processing success.
 	 */
-	public static function sync_order_coupons( $order_id ) {
+	public static function sync_order_coupons( $order ) {
```

Comment:
Since this is a public function it should have a check to ensure that the passed parameter is an order object.